pkg/destroy/aws/iamhelpers.go

```diff
@@ -107,6 +108,13 @@ func (search *IamUserSearch) arns(ctx context.Context) ([]string, error) {
 		func(results *iam.ListUsersOutput, lastPage bool) bool {
 			search.logger.Debugf("iterating over a page of %d IAM users", len(results.Users))
 			for _, user := range results.Users {
+				// By convention, IAM user names are created with the cluster name as prefix.
+				// We can cut down on expensive API calls by checking the prefix, before continuing
```

Comment:
I guess we can further speed up IAM resource search process by listing only user tags as the user query is only concerned about tags:

https://github.com/openshift/installer/blob/6667244d62f556c4a0c7194164985b4513ee77a3/pkg/destroy/aws/iamhelpers.go#L137-L146

We can use the API call: 

- https://docs.aws.amazon.com/IAM/latest/APIReference/API_ListUserTags.html
- AWS Go SDK func: [`ListUserTagsWithContext`](https://pkg.go.dev/github.com/aws/aws-sdk-go/service/iam#IAM.ListUserTagsWithContext)